src/geanyobject.c

```diff
@@ -380,3 +389,13 @@ GObject *geany_object_new(void)
 {
 	return g_object_new(GEANY_OBJECT_TYPE, NULL);
 }
+
+/** Get the global GeanyObject instance
+ *
+ * @return @transfer{none} The instance
+ */
+GEANY_API_SYMBOL
+GObject *geany_object_get_instance(void)
```

Comment:
Nope, I thought about this too. But pygobject's magic (that allows passing callbacks that fit to the signal's expected signature) only triggers for GObject.connect(). For all other functions the python function must match the signature of the C callback, but for signal handlers there is only GCallback regardless of the actual expected callback. `plugin_signal_connect_full` wouldn't help.